web/src/features/time/BottomSheet.ts

```diff
@@ -0,0 +1,340 @@
+export type SnapPoint = number;
+
+interface BottomSheetEvents {
+  onOpen?: () => void;
+  onClose?: () => void;
+  onSnap?: (snapIndex: number) => void;
+}
+
+export interface BottomSheetProps extends BottomSheetEvents {
+  snapPoints?: SnapPoint[];
+  backgroundColor?: string;
+  excludeElement?: HTMLElement;
+}
+
+function createBackdropElement(bgColor: string): HTMLDivElement {
+  const backdrop = document.createElement('div');
+  const hasTailwindBackground = bgColor.includes('bg-');
+
+  if (!hasTailwindBackground) {
+    Object.assign(backdrop.style, {
+      position: 'absolute',
+      left: '0',
+      right: '0',
+      bottom: '0',
+      backgroundColor: bgColor,
+      transition: 'height 0.3s ease-out',
+    });
+  }
+  if (hasTailwindBackground) {
+    Object.assign(backdrop.style, {
+      position: 'absolute',
+      left: '0',
+      right: '0',
+      bottom: '0',
+      transition: 'height 0.3s ease-out',
+    });
```

Comment:
Do we need both of these?

And even if we do it would be nice to deduplicate most of the code by only adding/changing the background color if tailwind is not used.